### PR TITLE
Updated url validation mechanism

### PR DESCRIPTION
**Pull request by iguana007**
Fixes issue, when absolute URL is being provided and contains port number.

### CODE REVIEW

File: nette.ajax.js [hunk 1]
```diff
@@ -329,7 +329,15 @@ $.nette.ext('validation', {
 
 		if (validate.url) {
 			// thx to @vrana
-			if (/:|^#/.test(analyze.form ? settings.url : analyze.el.attr('href'))) return false;
+			urlToValidate = analyze.form ? settings.url : analyze.el.attr('href');
```
Comment (vojtech-dobes) [inline nette.ajax.js]: `var urlToValidate`

**Comment by vojtech-dobes** [comment]
Could you please squash both commits into one? I would also remove comma from the body of first commit message as it should not be there I think :).

File: nette.ajax.js [hunk 2]
```diff
@@ -329,7 +329,15 @@ $.nette.ext('validation', {
...
 			// thx to @vrana
-			if (/:|^#/.test(analyze.form ? settings.url : analyze.el.attr('href'))) return false;
+			var urlToValidate = analyze.form ? settings.url : analyze.el.attr('href');
+			//Check if URL is absolute
+			if (/(?:^[a-z][a-z0-9+.-]*:|\/\/)/.test(urlToValidate)) {
+				//parse absolute URL
+				var parsedUrl = new URL(urlToValidate);
+				if (/:|^#/.test(parsedUrl['pathname'] + parsedUrl['search'] + parsedUrl['hash'])) return false;
+			} else {
+				if (/:|^#/.test(urlToValidate)) return false
```
Comment (vojtech-dobes) [inline nette.ajax.js]: semicolon `;` is missing at the end of line

File: nette.ajax.js [hunk 3]
```diff
@@ -329,7 +329,15 @@ $.nette.ext('validation', {
 
 		if (validate.url) {
 			// thx to @vrana
-			if (/:|^#/.test(analyze.form ? settings.url : analyze.el.attr('href'))) return false;
+			var urlToValidate = analyze.form ? settings.url : analyze.el.attr('href');
+			//Check if URL is absolute
```
Comment (vojtech-dobes) [inline nette.ajax.js]: Please put space after `//` (it's easy to get this right first time by checking already existing codebase).

**Comment by iguana007** [comment]
I didn`t know, how to make it in this PR, so I created new one ... https://github.com/vojtech-dobes/nette.ajax.js/pull/124

**Comment by uestla** [comment]
@iguana007 So maybe close this one? ;-)

Next time you can just push to the branch in your fork from which you created this Pull request - the changes will appear.

**Comment by iguana007** [comment]
Closed

**Pull request closed without merging**